de.dlr.sc.virsat.server/src/de/dlr/sc/virsat/server/dataaccess/FlattenedStructuralElementInstance.java

```diff
@@ -0,0 +1,264 @@
+/*******************************************************************************
+ * Copyright (c) 2020 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.server.dataaccess;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.common.command.CompoundCommand;
+import org.eclipse.emf.edit.command.AddCommand;
+import org.eclipse.emf.edit.command.RemoveCommand;
+import org.eclipse.emf.edit.command.SetCommand;
+
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.general.GeneralPackage;
+import de.dlr.sc.virsat.model.dvlm.roles.Discipline;
+import de.dlr.sc.virsat.model.dvlm.structural.StructuralElementInstance;
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+
+public class FlattenedStructuralElementInstance {
+
+	// API read only
+	private String uuid;
+	private String structuralElementFullQualifiedName;
+	private String parent;
+	
+	// API read and write
+	private String name;
+	private String description;
+	private String discipline;
+	
+	// API read all and write on existing SEIS
+	private List<String> superSeis;
+	private List<String> children;
+	private List<String> categoryAssignments;
+	
+	public FlattenedStructuralElementInstance() { }
+	
+	/**
+	 * Constructor to flatten an existing sei
+	 * @param sei the sei to flatten
+	 */
+	public FlattenedStructuralElementInstance(StructuralElementInstance sei) {
+		setUuid(sei.getUuid().toString());
+		setSeFullQualifiedName(sei.getType().getFullQualifiedName());
+		setParent(sei.getParent() != null ? sei.getParent().getUuid().toString() : null);
+		
+		setName(sei.getName());
+		setDescription(sei.getDescription());
+		if (sei.getAssignedDiscipline() != null)	{
+			setDiscipline(sei.getAssignedDiscipline().getUuid().toString());
+		} else {
+			setDiscipline(null);
+		}
+		
+		setSuperSeis(collectParentUuids(sei));
+		setChildren(collectChildUuids(sei));
+		setCategoryAssignments(collectCategoryAssignmentUuids(sei));
+	}
+	
+	private List<String> collectParentUuids(StructuralElementInstance sei) {
+		List<String> uuids = new ArrayList<String>();
+		for (StructuralElementInstance superSei : sei.getSuperSeis()) {
```

Comment:
I replaced them by a method _List<String> collectUuids(EList<? extends IUuid> iuuids)_. What do you mean by external? Should that method be moved in another class?

Reply:
Not sure where to put it, but I dont just see it in here.
For example the resolveProperties in FlattenedCategoryAssignment also uses this logic.
So I think the logic should definitely reside outside of FlattenedStructuralElementINstance

Reply:
Would it make sense to have an AFlattenedElement class for this, that also holds the uuid, because i think at least until now every Flattened... class has a uuid?